lib/puppet/file_serving/base.rb

```diff
@@ -81,6 +81,6 @@ def to_data_hash
   end
 
   def self.absolute?(path)
-    Puppet::Util.absolute_path?(path, :posix) or (Puppet.features.microsoft_windows? and Puppet::Util.absolute_path?(path, :windows))
+    Pathname.new(path.to_s).absolute?
```

Comment:
There's some subtly here... According to https://github.com/puppetlabs/puppet/commit/94f5f53244eddb80bc4e8cb58859d12624b2daa9, windows agents sometimes need to know if a file path is an absolute posix path, which means we can't just use `Pathname`. I need to try this out on a Windows VM to see if that is still a requirement, as lots of things have changed in file serving since that went in.